web/controller/utils/useReadControlData.js

```diff
@@ -61,12 +63,48 @@ export const useReadControlData = (controlId, controlType) => {
     [controlData]
   );
 
+  const uploadPictures = async pictures => {
+    alerts.withApiErrorHandling(async () => {
+      const presignedUrlsRes = await api.jsonHttpQuery(
+        HTTP_QUERIES.controlPicturesGeneratePresignedUrls,
+        {
+          json: { control_id: controlId, nb_pictures: pictures.length }
+        }
+      );
+      const presignedUrls = presignedUrlsRes["presigned-urls"];
+
+      await Promise.all(
+        pictures.map(async (picture, index) => {
+          const uploadRes = await fetch(presignedUrls[index], {
+            method: "PUT",
+            headers: { "Content-Type": "image/png" },
+            body: picture.file
+          });
+          // TODO: check status is 200
+          console.log("uploadRes", uploadRes);
+        })
+      );
+
+      const picturesResponse = await api.graphQlMutate(
+        CONTROLLER_READ_CONTROL_PICTURES,
+        { controlId },
+        { context: { nonPublicApi: true } }
+      );
+      setControlData({
+        ...controlData,
+        pictures: picturesResponse.data.controlData.pictures
+      });
+      alerts.success("Vos photos ont bien été sauvegardées", "", 5000);
```

Comment:
suggestion(non-blocking): most of the time, we use 6000 ; I think it's not enough for RGAA compliancy but at least we'll be consistent